examples/IMUKittiExampleGPS.cpp

```diff
@@ -0,0 +1,293 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file IMUKittiExampleGPS
+ * @brief Example of application of ISAM2 for GPS-aided navigation on the KITTI VISION BENCHMARK SUITE
+ * @author Ported by Thomas Jespersen (thomasj@tkjelectronics.dk), TKJ Electronics
+ */
+
+// GTSAM related includes.
+#include <gtsam/navigation/CombinedImuFactor.h>
+#include <gtsam/navigation/GPSFactor.h>
+#include <gtsam/navigation/ImuFactor.h>
+#include <gtsam/slam/dataset.h>
+#include <gtsam/slam/BetweenFactor.h>
+#include <gtsam/slam/PriorFactor.h>
+#include <gtsam/nonlinear/ISAM2.h>
+#include <gtsam/nonlinear/ISAM2Params.h>
+#include <gtsam/nonlinear/LevenbergMarquardtOptimizer.h>
+#include <gtsam/nonlinear/NonlinearFactorGraph.h>
+#include <gtsam/inference/Symbol.h>
+#include <cstring>
+#include <fstream>
+#include <iostream>
+
+using namespace gtsam;
+using namespace std;
+
+using symbol_shorthand::X; // Pose3 (x,y,z,r,p,y)
+using symbol_shorthand::V; // Vel   (xdot,ydot,zdot)
+using symbol_shorthand::B; // Bias  (ax,ay,az,gx,gy,gz)
+
+typedef struct {
+    double Time;
+    double dt;
+    Vector3 Accelerometer;
+    Vector3 Gyroscope; // omega
+} imuMeasurement_t;
+
+typedef struct {
+    double Time;
+    Vector3 Position; // x,y,z
+} gpsMeasurement_t;
+
+const string output_filename = "IMUKittyExampleGPSResults.csv";
+
+int main(int argc, char* argv[])
+{
+    string line;
```

Comment:
How about isolating reading of files in separate function?

Reply:
Addressed in b48bdfdc696328be3fedfe11db18b83d3341e7bc

Reply:
Addressed in b48bdfdc696328be3fedfe11db18b83d3341e7bc

Reply:
Addressed in b48bdfdc696328be3fedfe11db18b83d3341e7bc